polar-core/src/vm.rs

```diff
@@ -1529,16 +1530,35 @@ impl PolarVirtualMachine {
                     }
                     Value::Partial(partial) => {
                         let mut partial = partial.clone();
-                        if matches!(item.value(), Value::Variable(_) | Value::Partial(_)) {
-                            let item_partial = partial.in_unbound(item);
-                            self.bind(
-                                &item_partial.value().as_partial().unwrap().name().clone(),
-                                item_partial,
-                            );
-                            self.bind(partial.name(), partial.clone().into_term());
-                        } else {
-                            partial.in_contains(item);
-                            self.bind(&partial.name().clone(), partial.into_term());
+                        let partial_name = partial.name().clone();
+
+                        let partial_in_partial = |mut item: Partial, mut collection: Partial| {
+                            if collection.constraints().is_empty() {
```

Comment:
Why this branch?

Reply:
I think you're right about partials coming from the Django adapter, but we do have a test for it in the core (`test_in_partial`).